test_runner/regress/test_aux_files.py

```diff
@@ -0,0 +1,81 @@
+from fixtures.log_helper import log
+from fixtures.neon_fixtures import (
+    NeonEnv,
+    logical_replication_sync,
+)
+
+
```

Comment:
Given the complexity of configurability, should we have tests that one cannot go from CrossValidation or V2 to V1?

Should we have a test case which shows how should the configuration work together with branching?

Both of these could be rust level unit tests, I'm pretty sure. Those are run with `cargo nextest run -p pageserver`, and assume unique access to the statics.

Reply:
yep, the e2e test is used to verify the behavior that the flag gets changed only on the write path + it gets persisted. Added https://github.com/neondatabase/neon/pull/7668/commits/72cd2c389a6b8e801aabca02940330d3644ca644 to test all migration paths.

Reply:
aux flag on branching is tested with https://github.com/neondatabase/neon/pull/7668/commits/c369b89356a1b9fd3986787583aa514e1d4e4db3